lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -282,6 +294,17 @@ def get_last_edit(self, obj):
             last_edit["reason"] = EDIT_REASON_CODES.get(reason_code)
         return last_edit
 
+    def get_edit_by_label(self, obj):
+        """
+        Returns the role label for the last edit user.
+        """
+        is_user_author = str(obj['user_id']) == str(self.context['request'].user.id)
```

Comment:
why do we need to cast to str for the condition here?

Reply:
User id from request.user.id is integer whereas thread.user_id is in string

Reply:
we only need to convert one of those to str then?

Reply:
yes... just to be on safe side, I converted both (as the same has been done in other methods of serializer)